common/queue/src/main/java/org/thingsboard/server/queue/pubsub/TbPubSubProducerTemplate.java

```diff
@@ -120,7 +121,14 @@ private Publisher getOrCreatePublisher(String topic) {
             try {
                 admin.createTopicIfNotExists(topic);
                 ProjectTopicName topicName = ProjectTopicName.of(pubSubSettings.getProjectId(), topic);
-                Publisher publisher = Publisher.newBuilder(topicName).setCredentialsProvider(pubSubSettings.getCredentialsProvider()).build();
+                InstantiatingExecutorProvider executorProvider =
```

Comment:
Please check the builder uses the executor provider

![image](https://github.com/thingsboard/thingsboard/assets/79898499/fee29078-38b0-44f6-8179-5b1b23646946)

As far as I see,  the `InstantiatingExecutorProvider` will create a new thread pool all the time you create a producer.
I prefer to have only an instance of a thread pool.

![image](https://github.com/thingsboard/thingsboard/assets/79898499/43c284d1-f189-4291-b6fd-2d1c3dcb0025)

InstantiatingExecutorProvider

![image](https://github.com/thingsboard/thingsboard/assets/79898499/eb4b9214-04b3-4654-b286-f367fd70a5c8)

I believe the `FixedExecutorProvider` is a proper choice. The `ScheduledExecutorService` you probably should create and control yourself. You can found many examples in the codebase.

![image](https://github.com/thingsboard/thingsboard/assets/79898499/cc3cc48e-0eaa-45d8-a2c5-d65540968194)